apps/labs/posts/nebari-typer-cli.md

```diff
@@ -0,0 +1,170 @@
+     src="/posts/asmi-blogposts/nebari-cli-1.1.png" 
+    />
+    <br />
+    <img
+     alt="Output of nebari init --guided-init command"
+     src="/posts/asmi-blogposts/nebari-cli-gcp-1.2.png" 
+     />
+    <br />
+    <img
+     alt="Output of nebari init --guided-init command"
+     src="/posts/asmi-blogposts/nebari-cli-1.3.png" 
+     height="500"/>
+    <br />
+    <i>Output of nebari init guided-init command</i>
+</p>
+
+As we can see it's more like a survey that asks users to fill out the correct details for generating the `nebari-config.yaml` file. We have used the [questionary library](https://github.com/tmbo/questionary) for having the options for the questions.
+
+After creating or generating the `nebari-config.yaml`, It's time to validate and render it respectively. First we will see the `help` command output for the validate to see all the options and requirements available for it.
+
+```bash
+  nebari validate --help
+```
+
+<p align="center">
+    <img
+     alt="Output of nebari validate help"
+     src="/posts/asmi-blogposts/nebari-validate-help.png" />
+    <br />
+    <i>Output of nebari validate help command</i>
+</p>
+
+As we can see we have to pass the `-c/--config` flag for validating the `nebari-config.yaml` file and we have to do the same for rendering the infrastructure using the render command.
+
+```bash
+  nebari validate -c qhub-config.yaml && nebari render -c qhub-config.yaml
+```
+
+<p align="center">
+    <img
+     alt="Output of nebari validate and neabri render"
+     src="/posts/asmi-blogposts/nebari-cli-1.4.png" />
+    <br />
+    <i>Output of nebari validate and render command</i>
+</p>
+
+Render command builds the infrasture for the deployment. It creates all terraform stages file that are required during deployment.
+
+**_NOTE:_** It's not neccessary to run validate and render command seperately. Because it runs under the hood of the deploy command.
+
+Now we will see the ouput of the `nebari-deploy` help command. Deploy command creates the provisioned resources that Nebari requires and deploys your project. Nebari also has a destroy command that works the same way the deploy works but instead of creating the provisioned resources it destroys it. It comes with the confirmation that is very important feauture that the new CLI have because previously it didn't ask for the confirmation and the cluster gets destroyed right after executing it (we don't want that thing to happen with us even mistakenly).
+
+```bash
+  nebari deploy --help
+```
+
+<p align="center">
+    <img
+     alt="Output of nebari deploy help"
+     src="/posts/asmi-blogposts/nebari-cli-1.6.png" />
+    <br />
+    <i>Output of nebari deploy help command and destroy command</i>
+</p>
+
+It has many flag which user can pass while executing the deploy command according to it's preferences. Here also `-c` flag is important to pass to make the deployment of `nebari-config.yaml` file successful.
```

Comment:
```suggestion
The deploy command also has a `--config` flag.
```

I removed the sentence about additional flags here because one of the potential enhancements we want to make in the future is to remove `--dns-provider` and `--dns-auto-provision` and move them to the `init` stage. We just didn't have the time for it this time around.